core/logic.py

```diff
@@ -0,0 +1,74 @@
+from context import models
+from models import model
+
+dojo = None
+def init_app():
+    #create dojo
+    if not dojo:
+        return model.Dojo("Andela-Kenya")
+    else:
+        return dojo
+
+
+
+def create_room(room_type, room_name):
+    """
+    input : room_type -> string represent type of room_type
+    room_name -> string represent name of room_name
+            output : returns -> return Room with name -> room_name
+    Raises -> TypeError if room_name exists
+            'Invalid name ' if room_name exists
+    """
+    #remove excess white charcters
+    room_name_stripped =  room_name.strip()
+    room_type_stripped = room_type.strip()
+
+    if len(room_type_stripped) == 0:
+        raise TypeError
+    room_type_cleaned = room_type_stripped
+
+    if len(room_name_stripped) == 0:
+        return 'Invalid name'
+    room_name_cleaned = room_name_stripped
+
```

Comment:
Try and make your code more readable, at first glance I could not understand what was happening but now I see that you are trying to check if rooms are not being created using empty strings. I believe there are better ways to do this and also more important things to validate, like whether this room name exists.